src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -345,7 +359,12 @@ public String getFullName() {
      * @since 1.515
      */
     public String getFullDisplayName() {
-    	return TestNameTransformer.getTransformedName(getFullName());
+        String archiveId = this.parent != null ? Util.fixEmptyAndTrim(parent.getArchiveId()) : null;
```

Comment:
calling fixEmpty and trim repeatedly is not the best as the String will be searched for spaces repeatedly.
better to fixEmptyAndTrim when it is set in the parent and just then call parent.getArchiveId()